lib/AST/ASTContext.cpp

```diff
@@ -1540,10 +1540,12 @@ ASTContext::getXTUDefinition(const FunctionDecl *FD, CompilerInstance &CI,
             iterateContextDecls(TU, MangledFnName, MangleCtx)) {
     llvm::errs() << "Importing function " << MangledFnName << " from "
                  << ASTFileName << "\n";
+
     // FIXME: Refactor const_cast
     auto *ToDecl = cast<FunctionDecl>(
         Importer.Import(const_cast<FunctionDecl *>(ResultDecl)));
-    assert(ToDecl->hasBody());
+    if(!ToDecl || !ToDecl->hasBody())
```

Comment:
So it can happen if the importer does not work correctly and cannot import the given function. But i agree that this should not block the whole analysis. So make this a compilation parameter:
XTU_ALLOW_FAILED_IMPORT
#ifdef (XTU_ALLOW_FAILED_IMPORT)
assert(ToDecl->hasBody());
#else 
if(!ToDecl->hasBody())
#endif 

btw: ToDecl cannot be null as cast<FunctionDecl> is used.